ITensorGaussianMPS/src/gmps.jl

```diff
@@ -196,45 +489,81 @@ stored in a structure as a set of Givens rotations.
 
 The correlation matrix should be Hermitian, and will be treated as if it itensor
 in the algorithm.
+
+If `is_bcs`, the correlation matrix is assumed to be in interlaced format:
+Λ[2*i-1:2*i,2*j-1:2*j]=[[c_i c_j^dagger , c_i c_j ], [c_i^dagger c_j^dagger,c_i^dagger c_j]]
+Note that this may not be the standard choice in the literature, but it is internally
+consistent with the format of single-particle Hamiltonians and Slater determinants employed.
 """
+###Backward Compatibility
 function correlation_matrix_to_gmps(
-  Λ0::AbstractMatrix{ElT}; eigval_cutoff::Float64=1e-8, maxblocksize::Int=size(Λ0, 1)
-) where {ElT<:Number}
-  Λ = Hermitian(Λ0)
-  N = size(Λ, 1)
+  Λ0::AbstractMatrix; eigval_cutoff::Float64=1e-8, maxblocksize::Int=size(Λ0.data, 1)
+)
+  return correlation_matrix_to_gmps(
+    Λ0; eigval_cutoff=eigval_cutoff, minblocksize=1, maxblocksize=maxblocksize
+  )
+end
+
+function correlation_matrix_to_gmps(
+  Λ0::AbstractMatrix;
+  eigval_cutoff::Float64=1e-8,
+  minblocksize::Int=1,
+  maxblocksize::Int=size(Λ0.data, 1),
+)
+  return correlation_matrix_to_gmps(
+    ConservingNf(Λ0);
+    eigval_cutoff=eigval_cutoff,
+    minblocksize=minblocksize,
+    maxblocksize=maxblocksize,
+  )
+end
+
+function correlation_matrix_to_gmps(
+  Λ0::T;
+  eigval_cutoff::Float64=1e-8,
+  minblocksize::Int=1,
+  maxblocksize::Int=size(Λ0.data, 1),
+) where {T<:Union{Pairing,ConservingNf}}
+  Λ = T(Hermitian(copy(Λ0.data)))
+  ElT = eltype(Λ.data)
   V = Circuit{ElT}([])
-  ns = Vector{real(ElT)}(undef, N)
   err_tot = 0.0
-  for i in 1:N
-    blocksize = 0
-    n = 0.0
+  Λ = check_pairing_correlations(Λ)
+  N = size(Λ.data, 1)
+  calctype = typeof(Λ)
+  factor = calctype <: Pairing ? 2 : 1
```

Comment:
Could turn this into a generic function `site_stride(::ConservesNf) = 1`, `site_stride(::ConservesNfParity) = 2`.